plugins/topology/src/components/Graph/WorkloadNode.tsx

```diff
@@ -0,0 +1,194 @@
+import { usePodsWatcher } from '../../hooks/usePodsWatcher';
+import { K8sResourceKind } from '../../types/types';
+
+import './WorkloadNode.css';
+
+const POD_STATUS_NORMAL = 1;
+const POD_STATUS_WARNING = 2;
+const POD_STATUS_DANGER = 3;
+
+const StatusSeverities = {
+  danger: [
+    'ContainerCannotRun',
+    'CrashLoopBackOff',
+    'Critical',
+    'ErrImagePull',
+    'Error',
+    'Failed',
+    'Failure',
+    'ImagePullBackOff',
+    'InstallCheckFailed',
+    'Lost',
+    'Rejected',
+    'UpgradeFailed',
+  ],
+  warning: [
+    'Cancelled',
+    'Deleting',
+    'Expired',
+    'Not Ready',
+    'Terminating',
+    'Warning',
+    'RequiresApproval',
+  ],
+  normal: [
+    'New',
+    'Pending',
+    'Planning',
+    'ContainerCreating',
+    'UpgradePending',
+    'In Progress',
+    'Installing',
+    'InstallReady',
+    'Replacing',
+    'Running',
+    'Updating',
+    'Upgrading',
+    'Accepted',
+    'Active',
+    'Bound',
+    'Complete',
+    'Completed',
+    'Created',
+    'Enabled',
+    'Succeeded',
+    'Ready',
+    'Up to date',
+    'Provisioned as node',
+    'Preferred',
+    'Connected',
+    'Info',
+    'Unknown',
+    'PipelineNotStarted',
+  ],
+};
+
+export const getNodePodStatus = (podStatus: AllPodStatus): number => {
+  switch (podStatus) {
+    case AllPodStatus.Failed:
+    case AllPodStatus.CrashLoopBackOff:
+      return POD_STATUS_DANGER;
+    case AllPodStatus.Warning:
+      return POD_STATUS_WARNING;
+    default:
+      return POD_STATUS_NORMAL;
+  }
+};
+
+export const getAggregateStatus = (
+  donutStatus: any,
+  pipelineStatus: string,
+): NodeStatus => {
+  const worstPodStatus =
+    donutStatus?.pods?.reduce((acc: any, pod: any) => {
+      return Math.max(acc, getNodePodStatus(getPodStatus(pod)));
+    }, POD_STATUS_NORMAL) ?? NodeStatus.default;
+
+  if (
+    worstPodStatus === POD_STATUS_DANGER ||
+    StatusSeverities.danger.includes(pipelineStatus)
+  ) {
+    return NodeStatus.danger;
+  }
+  if (
+    worstPodStatus === POD_STATUS_WARNING ||
+    StatusSeverities.warning.includes(pipelineStatus)
+  ) {
+    return NodeStatus.warning;
+  }
+  return NodeStatus.default;
+};
+
+type InnerWorkloadNodeProps = {
+  element: Node;
+} & Partial<WithSelectionProps & WithDragNodeProps>;
+
+const InnerWorkloadNode: React.FC<InnerWorkloadNodeProps> = observer(
+  ({ element, ...rest }) => {
+    const resource = getTopologyResourceObject(
+      element.getData(),
+    ) as K8sResourceKind;
+    const { podData, loadError, loaded } = usePodsWatcher(
+      resource,
+      resource.kind,
+    );
+    const donutStatus = loaded && !loadError ? podData : null;
+    const { width, height } = element.getDimensions();
+    const workloadData = element.getData().data;
+    const [hover, hoverRef] = useHover();
+    const size = Math.min(width, height);
+    const { radius, decoratorRadius } = calculateRadius(size);
+    const cx = width / 2;
+    const cy = height / 2;
```

Comment:
```suggestion
    const cx = width / 2;
    const cy = height / 2;
```

can you please use a variable which is more readable